users/models.py

```diff
@@ -1,50 +1,69 @@
+    PermissionsMixin
+)
+from users.enums import UserRole
 
+class UserManager(BaseUserManager):
+    def create_user(self, email=None, phone=None, password=None, **extra_fields):
+        if not email and not phone:
+            raise ValueError("A user must have at least an email or phone number.")
+        
+        email = self.normalize_email(email) if email else None
+        
+        user = self.model(email=email, phone=phone, **extra_fields)
+        user.set_password(password)
+        user.save(using=self._db)
+        return user
 
-# Custom user model (replaces Django's default User)
-class CustomUser(AbstractUser):
-    """
-    CustomUser model
+    def create_superuser(self, email, phone, password=None, **extra_fields):
+        extra_fields.setdefault('is_staff', True)
+        extra_fields.setdefault('is_superuser', True)
 
-    Adds details:
-        phone number
-        role : Admin, Treasurer, User
-    """
+        
+        if not extra_fields["is_staff"]:
+            raise ValueError("Superuser must have is_staff=True.")
+        if not extra_fields["is_superuser"]:
+            raise ValueError("Superuser must have is_superuser=True.")
 
-    ADMIN = 'Admin'
-    TREASURE = 'Treasure'
-    USER = 'User'
-
-    ROLE_CHOICES = [
-        (ADMIN, 'admin'),
-        (USER, 'user'),
-        (TREASURE, 'treasure')
-    ]
-
-    email = models.EmailField(unique=True)
-    phone_number = models.CharField(max_length=13, blank=True)
-    #national_id = models.CharField(max_length=12, blank=True)
-    role = models.CharField(
-        max_length=10,
-        choices=ROLE_CHOICES,
-        default='user',
-    )
-
-    def __str__(self):
-        return f"{self.username} ({self.role})"
+        return self.create_user(email, phone, password, **extra_fields)
     
-class UserProfile(models.Model):
-    """
-    User profile model
+class User(AbstractBaseUser, PermissionsMixin):
+        id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+        first_name = models.CharField(max_length=30, blank=True)
+        last_name = models.CharField(max_length=30, blank=True)
+        email = models.EmailField(unique=True, null=True, blank=True)
+        phone = models.CharField(max_length=15, unique=True, null=True, blank=True)
+        
+        is_active = models.BooleanField(default=True)
+        is_staff = models.BooleanField(default=False)
+        role = models.CharField(
+            max_length=20,
+            choices=UserRole.choices(),
+            default=UserRole.MEMBER.value
+        )
+        created_at = models.DateTimeField(auto_now_add=True)
 
-    Adds details:
-        bio
-        avatar
-    """
-    user = models.OneToOneField(CustomUser, on_delete=models.CASCADE, related_name='profile')
+        USERNAME_FIELD = 'email'
+        REQUIRED_FIELDS = []
+        
+        objects = UserManager()
+        
+        def __str__(self):
+            return self.email or self.phone or self.name
```

Comment:
The `User` model doesn't have a `name` field, so using `self.name` won't work. Please use a combination of `self.first_name` and `self.last_name` instead